ui/components-react/src/components-react/editors/EditorContainer.tsx

```diff
@@ -218,91 +259,43 @@ export class EditorContainer extends React.PureComponent<EditorContainerProps> {
           errorMessage.messageType
         );
     }
-  }
-
-  private async isNewValueValid(value: PropertyValue): Promise<boolean> {
-    let isValid = true;
-
-    // istanbul ignore else
-    if (this._propertyEditor && this.props.propertyRecord) {
-      const validateResult = await this._propertyEditor.validateValue(
-        value,
-        this.props.propertyRecord
-      );
-
-      if (validateResult.encounteredError) {
-        this.displayOutputMessage(validateResult.errorMessage);
-        isValid = false;
-      }
-    } else {
-      isValid = false;
-    }
-
-    return isValid;
-  }
-
-  private _handleEditorCommit = (args: PropertyUpdatedArgs): void => {
-    void this._commit(args);
   };
 
-  private _handleContainerCommit = async (): Promise<void> => {
-    const newValue =
-      this._editorRef && (await this._editorRef.getPropertyValue());
-    // istanbul ignore else
-    if (newValue !== undefined) {
-      void this._commit({
-        propertyRecord: this.props.propertyRecord,
-        newValue,
-      });
-    }
-  };
-
-  private _commit = async (args: PropertyUpdatedArgs) => {
+  const commit = async (args: PropertyUpdatedArgs) => {
     const newValue = args.newValue;
-    const isValid = await this.isNewValueValid(newValue);
+    const isValid = await isNewValueValid(newValue);
     if (isValid) {
       let doCommit = true;
       // istanbul ignore else
-      if (this._propertyEditor && args.propertyRecord) {
-        const commitResult = await this._propertyEditor.commitValue(
+      if (propertyEditor && args.propertyRecord) {
+        const commitResult = await propertyEditor.commitValue(
           newValue,
           args.propertyRecord
         );
         if (commitResult.encounteredError) {
-          this.displayOutputMessage(commitResult.errorMessage);
+          displayOutputMessage(commitResult.errorMessage);
           doCommit = false;
         }
       }
 
       if (doCommit) {
-        this.props.onCommit(args);
+        onCommit(args);
       }
     }
   };
 
-  private _handleEditorCancel = () => {
-    this._commitCancel();
-  };
-
-  private _commitCancel = () => {
-    this.props.onCancel();
-  };
-
-  /** @internal */
-  public override render() {
-    return (
-      <span
-        className="components-editor-container"
-        onBlur={this._handleContainerBlur}
-        onKeyDown={this._handleKeyDown}
-        onClick={this._handleClick}
-        onContextMenu={this._handleRightClick}
-        title={this.props.title}
-        data-testid="editor-container"
-        role="presentation"
-      >
-        {this.createEditor()}
-      </span>
-    );
-  }
+  return (
+    <span
+      className="components-editor-container"
+      onBlur={handleContainerBlur}
+      onKeyDown={handleKeyDown}
+      onClick={handleClick}
+      onContextMenu={handleClick}
+      title={title}
+      data-testid="editor-container"
+      role="presentation"
+    >
+      {createEditor()}
```

Comment:
Remove this function call, inline it in the component body instead.

Reply:
DO you mean inline the whole editor creation? that seem excessive as its 30+ lines.

Reply:
It's 30lines + function declaration (that's used once) either way.

Reply:
thas a good point! I just wanted to double check before I made that change